siliconcompiler/utils/__init__.py

```
@@ -580,6 +592,329 @@ def print_traceback(logger: logging.Logger, exception: Exception):
+
+    Args:
+        logger (logging.Logger): Logger used to emit warnings.
+    """
+    try:
+        installed = _installed_distribution_versions()
+        for pyproject in _find_editable_pyproject_paths():
+            data = _load_pyproject_data(pyproject)
+            if data is None:
+                continue
+            _check_project_dependencies(data, installed, logger)
+    except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)